.github/workflows/go.yml

```diff
@@ -94,23 +77,9 @@ jobs:
     - name: Check-out code
       uses: actions/checkout@v3
     - name: Set up Go using version from go.mod
-      uses: actions/setup-go@v3
+      uses: actions/setup-go@v4
```

Comment:
you need to set `cache-dependency-path` here, given that the relevant go.mod is not at the root of the repository

refer to https://github.com/actions/setup-go#caching-dependency-files-and-build-outputs